### PR TITLE
Api serivces

### PR DESCRIPTION
**Pull request by jakejakearell**
### What?
 * Consume Movie API for most popular user movies
 * Create Discover Page
 * Start Movie Page
 * Add Figaro
### Testing?
 * Feature testing in RSpec

### CODE REVIEW

File: app/controllers/discover_controller.rb [hunk 1]
```diff
@@ -1,3 +1,14 @@
 class DiscoverController < ApplicationController
-  def index; end
+  before_action :require_current_user
+
+  def index
+    # @films = FilmSearch.new.top_40_films if params[:button_clicked]
```
Comment (Gvieve) [inline app/controllers/discover_controller.rb]: If there is unused code we should remove it before opening a PR

File: spec/features/discover/index_spec.rb [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+  end
+
+  it "I see a welcome message with the name of the user" do
+    visit root_path
+
+    fill_in "email", with: @user_1.email
+    fill_in "password", with: 'sassyperson1'
+    click_button "Sign In"
+
+    expect(current_path).to eq(dashboard_index_path)
```
Comment (Gvieve) [inline spec/features/discover/index_spec.rb:15]: Let's implement the mocked authenticated user for these tests where necessary

File: spec/rails_helper.rb [hunk 3]
```diff
@@ -9,6 +9,8 @@
 VCR.configure do |config|
   config.cassette_library_dir = "fixtures/vcr_cassettes"
   config.hook_into :webmock
+  # config.configure_rspec_metadata!
+  # config.response.body.force_encoding('UTF-8')
```
Comment (Gvieve) [inline spec/rails_helper.rb:14]: We should implement the keep api key private before merging this to main and re-record any cassettes

**Review by Gvieve — CHANGES_REQUESTED** [review]
Please look at my comments for suggested changes

File: spec/features/discover/index_spec.rb (hunk 2 above)
Comment (harrison-blake) [reply spec/features/discover/index_spec.rb:15]: DONE

File: app/controllers/discover_controller.rb (hunk 1 above)
Comment (harrison-blake) [reply app/controllers/discover_controller.rb]: Thanks I got rid of it!

File: spec/rails_helper.rb (hunk 3 above)
Comment (harrison-blake) [reply spec/rails_helper.rb:14]: thanks it is done

**Review by Gvieve — APPROVED** [review]
Thanks for making the changes. This looks good. Amazing work!

**Pull request merged**